### DISCUSSION TITLE
Freezes on Emacs 30

### DISCUSSION THREAD

**Discussion by paaguti** [minad/jinx#149] [Q&A]
Hi,

I was using jinx until it started to produce hiccups and freezes of more that two seconds in my emacs setup.

Thx, /PA

**Comment by paaguti** [comment]
Also happens with jinx1.3 from elpa.gnu.org. Mainly with org-mode files...

**Comment by minad** [comment]
Thanks for the report. Can you please give a minimal recipe based on emacs -Q which demonstrates the issue? Which Emacs version do you use? Maybe you can also try different Jinx versions to check if the problem is a new occurrence? Note that many people use Org with Jinx so we should see the problem coming up more often, in case it is really caused by Jinx.

**Comment by paaguti** [comment]
OK,

put this in `test.el`

```
(defun activate-jinx()
  (interactive)
  (jinx-languages "en_GB")
  (jinx-mode +1))
```
open any longer org-mode file with
```
emacs -Q -l ./test.el <org-file>
```
Move around and edit the file (without problems or hiccups)

`M-x activate-jinx`

on the org buffer and the hiccups begin. Emacs will start freezing for 1+ seconds 
sometimes when moving the cursor.

**Reply by minad** [reply]
That's odd. Is this on Emacs 30? Jinx works flawlessly on 29.2, so there must be some upstream change causing this regression.

**Reply by minad** [reply]
Related reports on org-devel and the Emacs bug tracker by @paaguti:

* https://lists.gnu.org/archive/html/emacs-orgmode/2024-03/msg00111.html
* https://lists.gnu.org/archive/html/bug-gnu-emacs/2024-03/msg00335.html

It seems there exists some interaction issue between Jinx and Org since Emacs 30. cc @yantar92 Are you aware of any changes to the text property, overlay or display engine machinery, which could be causing this? I am on Emacs 29.2 (with the newest Org development version from ELPA-devel) and there Jinx works without issues in very large Org files like [shakespeare.org](https://github.com/jdtsmith/shakespeare.org/blob/main/shakespeare.org), with both `org-fold-core-style=overlays` and `org-fold-core-style=text-properties`. Jinx maintains its own text properties to keep track of already spell-checked regions, but these text properties should not interfere with Org.

(Btw, in my tests `org-fold-core-style=text-properties` is still more performant even on Emacs 29.)

**Comment by yantar92** [comment]
Daniel Mendler ***@***.***> writes:

> Related reports on org-devel and the Emacs bug tracker by @paaguti:
>
> * https://lists.gnu.org/archive/html/emacs-orgmode/2024-03/msg00111.html
> * https://lists.gnu.org/archive/html/bug-gnu-emacs/2024-03/msg00335.html
>
> It seems there exists some interaction issue between Jinx and Org
> since Emacs 30. cc @yantar92 Are you aware of any changes to the text
> property or overlay machinery, which could be causing this? I am on
> Emacs 29.2 and there Jinx works without issues in very large Org files
> like [shakespeare.org](https://github.com/jdtsmith/shakespeare.org),
> with both `org-fold-core-style=overlays` and
> `org-fold-core-style=text-properties`.

I am unable to reproduce using the latest Org mode main + the latest
Emacs master. I am also unable to reproduce using Emacs 29.

> (Btw, in my tests org-fold-core-style=text-properties is still more
> performant even on Emacs 29.)

Feel free to report the profile or reproducer. The reason I decided to
switch back to overlays is that it is pretty much impossible to
support isearch when using text properties. (I tried, and gave up)

-- 
Ihor Radchenko // yantar92,
Org mode contributor,
Learn more about Org mode at <https://orgmode.org/>.
Support Org development at <https://liberapay.com/org-mode>,
or support my work at <https://liberapay.com/yantar92>

**Reply by minad** [reply]
> I am unable to reproduce using the latest Org mode main + the latest
Emacs master. I am also unable to reproduce using Emacs 29.

Thank you very much. Then this must be an issue in @paaguti's setup. @paaguti, maybe you can try a newer commit of Emacs? It could be that you are on a bad commit. There have been changes to the display engine recently. I close this until we get a better reproducer.

> The reason I decided to
switch back to overlays is that it is pretty much impossible to
support isearch when using text properties. (I tried, and gave up)

Yes, this makes sense. For me the text-property based folding works acceptably well (with both Isearch and my `consult-line`). For the `consult-line` command (basically an interactive variant of `occur`) I had to write special code to support unfolding/refolding, see https://github.com/minad/consult/blob/888b4525361c4050f35f4766fe7651c97d3274ce/consult.el#L1444.
